### ISSUE TITLE
Allow switching to a full featured browser

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/focus-android#26]
Currently we just have a share action but no way to switch to an actual browser.

**Comment by pocmo** [comment]
@antlam We need to define how this should look like.

On new Android versions you can get the list of browser anymore if the user has a default browser set: We only get the default browser. However if Focus actually *is* the default browser then this doesn't help us.

However we can look which browsers from a list of known browsers are installed and then offer to open in one of those browsers. I would love to see something like our quick sharing feature in Fennec: First offer all browsers, then later allow quick access to the most used one. And do we want to promote Firefox here somehow? E.g. offer to install it if it's not yet.

**Comment by antlam** [comment]
We should make sure we're switching to Private Browsing in Firefox. That's a change we made in iOS after some user feedback.

I feel like the UI for this should go in our Overflow menu... like the latest mock in #52 ... thoughts?

(After the -> and 'refresh' buttons) perhaps something like:

- Share...
- Open in Firefox (Private Browsing)
- Open in _Default browser_ 
- Settings

Where the "Default browser" option will only show up if it's not Firefox or Focus? And the "Firefox (Private Browsing)" link will lead to the GPS install page if not installed?

**Comment by bbinto** [comment]
"We should make sure we're switching to Private Browsing in Firefox. That's a change we made in iOS after some user feedback."
--> Can you point me to that feedback, I thought this was something some of the team decided. 

I wanted to post here: https://github.com/mozilla-mobile/focus-android/issues/52 but thought it might fit better here:

How do you change the default browser though if it's set to Firefox or Focus, would there be no "Open with ..." option?

I think we can/should be proud of our cross-promotion and give Firefox a more prominent placeholder in the DOT menu. (that's the whole point of app diversification :))

- We might not need the Back button (if the device back button works, doesn't currently work)
- Forward
- Refresh
- Share

a) Default browser == Firefox
     Open with Firefox 
    "Open with ..." (I think we need this, otherwise the user is stuck with Firefox, unless they reset app Settings on their phone)
b) Default browser == Focus
    Open with ... (to overwrite)
c) Not Firefox/Focus as Default
    Open with Firefox ( if Firefox is not installed, link to Google Play Store)
    Open with _Default browser_

- Settings

For first version, we can/should keep it simple, add telemetry to help us understand how often users actually pop-out.

**Comment by antlam** [comment]
Sorry! but this issue is getting a bit hard to follow for me. So I'm going to focus on the "switching to a full featured browser" user story here. Feel free to let me know if I'm missing the point here.

> As a user, I want a way to open the current URL in another browser 

With that, I'd suggest we add an item in the overflow menu - "Open in Firefox". This will be there regardless of what their default browser app is set to. 

If Firefox is installed, this button opens the current link in Private Browsing, in Firefox
If Firefox is **not installed**, this button opens the GPS for the user to download Firefox. 

Dealing with default browsers set by the user is kind of tricky here. I think the simplest solution would be to offer **another item** in the menu - "Open in _Default_". 

Even with that, I am slightly worried there's a moment of suspicion in the users mind. After all, we are surfacing their personal/private choice to them (their default browser of choice) and it might undermine our privacy story. 

Thoughts?

**Comment by antlam** [comment]
<img width="386" alt="screen shot 2017-02-17 at 2 42 54 pm" src="https://cloud.githubusercontent.com/assets/1994820/23085936/67e78768-f51f-11e6-80dd-46c0a1dc0902.png">

^ attaching visuals to help the convo

**Comment by ahunt** [comment]
FWIW chrome custom tabs already surfaces "Open in <your default browser>". (I'm guessing we'll want to use a very similar UI if we implement custom tabs ourselves, and since custom tabs seem to be quite popular, users might be used to "Open in X".)

**Comment by pocmo** [comment]
Custom tabs is a bit different. Open in X always links to the (full) default browser. In the case of Chrome this would be Chrome itself (the full version). However if Focus is the default browser then it doesn't make sense to link to itself. :)

**Comment by bbinto** [comment]
The app should be easy to use, right from the start, especially if they encounter a scenario where they need to pop-out. Having their preferred browser of choice listed seems reasonable to me, and becomes less of a privacy issue.

Anthony, your suggestion would be to always show Open with Firefox as the first option, followed by Open with _default browser_ unless Focus or Firefox is the default browser, correct?
As soon as they change their default browser (not possible inside Focus as per your suggestion), it would appear in the list of options below "Open with Firefox". 

That sounds like an acceptable UI behaviour to me.

**Comment by antlam** [comment]
Talked to @pocmo about this some more.

We're going to also have a "Open with..." option to help address some of these concerns. That should address all of them actually.

It'll look like something like this: 
Where the user has Chrome set as their default.

![screen shot 2017-03-02 at 12 01 18 pm](https://cloud.githubusercontent.com/assets/1994820/23524558/05ce44b0-ff40-11e6-8d5f-0dac8333fcfb.png)
 
If the user has Focus set as their default, then the Open in Chrome option just wouldn't appear.

**Comment by bbinto** [comment]
If UX finds the amount of "Open with" choices not too extensive, I'm ok with it.
